client/lib/media/actions.js

```diff
@@ -143,9 +143,6 @@ function uploadFiles( uploader, files, siteId ) {
 		// Generate a fake transient item that can be used immediately, even
 		// before the media has persisted to the server
 		const transientMedia = { date, ...MediaUtils.createTransientMedia( file ) };
-		if ( file.ID ) {
```

Comment:
This prevents the ID to be something we don't control: it will always be `media-XXX`, which our pipeline knows how to process. It removes case 3.